File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.length();
        int left = 0;
        int maxLength = 0;
        int len = 0;
        std::map<char,int> window;

        for(int right=0; right < n; right++)
        {
            while(window[s[right]] == 1)
            {
                window[s[left]] = 0;
                len--;
                left++;
            }

            window[s[right]] = 1;
            len++;

            if(len > maxLength) maxLength = len;
        }
        

        return maxLength;
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0

        maxLength = self.BFS(root, 0, 0)
    
        return maxLength

    def BFS(self, node, length, maxLength):
        length += 1

        if length > maxLength: maxLength = length

        if node.left:
            maxLength = self.BFS(node.left, length, maxLength)
        
        if node.right:
            maxLength = self.BFS(node.right, length, maxLength)

        length -= 1

        return maxLength
        